src/components/ha-bar.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+import { classMap } from "lit-html/directives/class-map";
+
+@customElement("ha-bar")
+export class HaBar extends LitElement {
+  @property({ type: Number, attribute: "max-value" }) public maxValue = 100;
+
+  @property({ type: Number }) public value!: number;
+
+  @property({ type: Number }) public target!: number; // Number between 0 and 100
+
+  protected render(): TemplateResult {
+    const valuePrecentage = this._calculateValuePrecentage();
+
+    return html`
+      <slot></slot>
+      <div class="bar">
+        <div class="value">
+          ${valuePrecentage}%
+        </div>
+        <svg width="150">
+          <g>
+            <rect width="100%" height="12"></rect>
+            <rect
+              class="${classMap({
+                "target-reached": this.target && valuePrecentage >= this.target,
+              })}"
+              width="${valuePrecentage}%"
+              height="12"
+              rx="4"
+            ></rect>
+          </g>
+        </svg>
+      </div>
+    `;
+  }
+
+  private _calculateValuePrecentage(): number {
```

Comment:
Should this be wrapped in memoize?

Reply:
Let's move these to a helper and use them here and in gauge <https://github.com/home-assistant/frontend/blob/dev/src/components/ha-gauge.ts#L19-L30>

Reply:
As is, or should I wrap it too?